src/nbl/builtin/builtinHeaderGen.py

```diff
@@ -1,94 +1,119 @@
-    outp.write("#endif // __INTELLISENSE__\n")
+#ifdef __INTELLISENSE__
+#include <codeanalysis\warnings.h>
+#pragma warning( push )
+#pragma warning ( disable : ALL_CODE_ANALYSIS_WARNINGS )
+#endif // __INTELLISENSE__
 
-    outp.write("#include <stdlib.h>\n")
-    outp.write("#include <cstdint>\n")
-    outp.write("#include <string>\n")
-    outp.write("#include <unordered_map>\n")
-    outp.write("#include <utility>\n#include <nbl/core/string/StringLiteral.h>\n\n")
+#include <stdlib.h>
+#include <cstdint>
+#include <string>
+#include <unordered_map>
+#include <utility>
+#include <chrono>
+#include <array>
+#include <nbl/core/string/StringLiteral.h>
+    """)
     
     if isSharedLibrary:   
-        outp.write("#if defined(__NBL_BUILDING_TARGET__) // currently compiling the target, this define is passed through the commandline\n")
-        outp.write("#if defined(_MSC_VER)\n")
-        outp.write("#define NBL_BR_API __declspec(dllexport)\n")
-        outp.write("#elif defined(__GNUC__)\n")
-        outp.write('#define NBL_BR_API __attribute__ ((visibility ("default")))' + "\n")
-        outp.write("#endif\n")
-        outp.write("#else\n")
-        outp.write("#if defined(_MSC_VER)\n")
-        outp.write("#define NBL_BR_API __declspec(dllimport)\n")
-        outp.write("#else\n")
-        outp.write("#define NBL_BR_API\n")
-        outp.write("#endif\n")
-        outp.write("#endif\n\n")
-    
-    outp.write("namespace " + resourcesNamespace + " { \n")
-    
-    if isSharedLibrary:
-        outp.write("\t\tNBL_BR_API std::pair<const uint8_t*, size_t> get_resource_runtime(const std::string& filename);\n\n")
-    else:
-        outp.write("\t\tstd::pair<const uint8_t*, size_t> get_resource_runtime(const std::string& filename);\n\n")
+        outp.write(f"""
+    #if defined(__NBL_BUILDING_TARGET__) // currently compiling the target, this define is passed through the commandline
+        #if defined(_MSC_VER)
+            #define NBL_BR_API __declspec(dllexport)
+        #elif defined(__GNUC__)
+             #define NBL_BR_API __attribute__ ((visibility ("default")))
+    #endif
+    #else
+        #if defined(_MSC_VER)
+            #define NBL_BR_API __declspec(dllimport)
+        #else
+            #define NBL_BR_API
+        #endif
+    #endif
+        """)
+        
+    outp.write(f"""
+namespace {resourcesNamespace}
+{{
     
-    outp.write("\t\ttemplate<nbl::core::StringLiteral Path>\n")
-    outp.write("\t\tconst std::pair<const uint8_t*, size_t> get_resource();\n")
+    struct SBuiltinFile
+    {{
+        const uint8_t* contents = nullptr;
+        size_t size = 0u;
+        std::array<uint64_t, 4> xx256Hash;
+        std::tm modified = {{ // Absolute time since 1970 in UTC+0
+            .tm_sec = 6,
+            .tm_min = 9,
+            .tm_hour = 6,
+            .tm_mday = 9,
+            .tm_mon = 6,
+            .tm_year = 9,
+            .tm_isdst = 0
+        }};
+    }};
```

Comment:
no reason for every archive namespace to re-define this struct, slap it in `nbl::system` somewhere

Reply:
resolved